packages/commons/src/auth/jwt.ts

```diff
@@ -22,53 +39,62 @@ export const readAuthDataFromJwtToken = (jwtToken: string): AuthData => {
   }
 };
 
-const getKey =
-  (
-    clients: jwksClient.JwksClient[],
-    logger: Logger
-  ): ((header: JwtHeader, callback: SigningKeyCallback) => void) =>
-  (header, callback) => {
-    // eslint-disable-next-line functional/no-let
-    let responseReceived = 0;
-    for (const client of clients) {
-      client.getSigningKey(header.kid, function (err, key) {
-        responseReceived = responseReceived + 1;
-        if (err && responseReceived === clients.length) {
-          logger.error(`Error getting signing key: ${err}`);
-          return callback(err, undefined);
-        } else if (!err) {
-          return callback(null, key?.getPublicKey());
-        }
-      });
+const getKey = async (
+  clients: jwksClient.JwksClient[],
+  kid: string,
+  logger: Logger
+): Promise<Secret> => {
+  logger.info(`Getting signing key for kid ${kid}`);
+  for (const client of clients) {
+    try {
+      const signingKey = await client.getSigningKey(kid);
+      return signingKey.getPublicKey();
+    } catch (error) {
+      // Continue to the next client
+      logger.info(`Skip Jwks client`);
     }
-  };
+  }
+
+  logger.error(`Error getting signing key`);
+  return Promise.reject();
```

Comment:
Since this function is try-caught where it's used, here you could `throw` instead of `Promise.resolve`

Reply:
Thrown with commit [6442b06](https://github.com/pagopa/interop-be-monorepo/pull/1002/commits/6442b06777a7158c98a9904957641d6959daea9d) 😜